docs/HowToUseSmartFS.md

```diff
@@ -0,0 +1,187 @@
+# How to use SmartFS
+
+SmartFS is FileSystem used for user partition in TizenRT.
+
+SmartFS stands for Sector Mapped Allocation for Really Tiny (SMART) flash.  
+It is a filesystem that has been designed to work primary with small,
+serial NOR type flash parts that are 1M byte to 16M byte in size
+(though this is not a limitation).
+
+The filesystem operates by segmenting the flash (or flash partition)
+into “logical sectors” of equal size and then managing them (allocating,
+mapping, chaining, releasing, etc.) to build files and directories.
+
+This guide explains the steps and details for enabling SmartFS on a new board.
+
+## SmartFS Code Layering
+
+Smart FS consists of two layers built on top of standard MTD Driver.  
+Below diagram shows the dependant layers part of Smart FS.
+
+```
++----------------------------------------------+
+| SMART FS (Smart FileSystem Layer)            |
++----------------------------------------------+
+| SMART MTD Driver (Smart Block Driver Layer)  |
++----------------------------------------------+
+| MTD Driver  (MTD Block Driver Layer)         |
++----------------------------------------------+
+| FLASH Driver (Flash Driver)                  |
++----------------------------------------------+
+```
+
+The code directly above the MTD driver is the SMART MTD layer.  
+This interfaces with the MTD (flash driver) layer and handles low-level
+media operations such as logical sector allocation, freeing and management,
+erase block management, low-level formatting, wear leveling, etc.
+
+On top of the SMART MTD layer is the Smart Filesystem Layer.  
+The SmartFS code uses the logical sector services of the SMART MTD layer to
+provide file and directory level management, such as creating new files,
+chaining logical sectors together to create files, creating directories and
+ file / directory search and management routines.
+
+## FLASH Driver
```

Comment:
How about having this explanation in the reverse order. as Each layer makes use of the services provided by underlying layer

Reply:
It is written in the order of which one must be brought up first during bring up phase.